todo_app/app.py

```diff
@@ -1,54 +1,120 @@
 import os
+from flask_login.utils import login_required
 import requests
-from flask import Flask, render_template, request, session, redirect, url_for
+from oauthlib.oauth2 import WebApplicationClient
+from flask import Flask, render_template, request, session, redirect, url_for, abort, current_app
 import todo_app.view_model as view_model
 from todo_app.Task import Task
+from todo_app.user import User
 import todo_app.trello as trello
+from todo_app.mongo_db_tasks import TasksDb
+from flask_login import LoginManager, UserMixin, login_required, login_user, logout_user, current_user
 
-app = Flask(__name__)
-headers = {
-   "Accept": "application/json"
-}
+taskDb = TasksDb()
+secret_key = os.environ.get('SECRET_KEY', 'secret_key')
+git_client_id = os.environ.get('GIT_CLIENT_ID')
+git_client_secret = os.environ.get('GIT_CLIENT_SECRET')
+admin_user_id = os.environ.get('APP_ADMIN_USER_ID', '000')
+login_manager = LoginManager()
+client = WebApplicationClient(git_client_id)
+ROLE = 'readwrite'
```

Comment:
Rather than doing most of this setup here, I'd suggest doing it inside the create_app function - we should only gather and create these instances (e.g. login_manager and client) when we actually create the app